fingerprint2.js

```diff
@@ -21,1431 +21,1376 @@
+        'RealPlayer',
+        'RealPlayer.RealPlayer(tm) ActiveX Control (32-bit)',
+        'RealVideo.RealVideo(tm) ActiveX Control (32-bit)',
+        'Scripting.Dictionary',
+        'SWCtl.SWCtl', // ShockWave player
+        'Shell.UIHelper',
+        'ShockwaveFlash.ShockwaveFlash', // flash plugin
+        'Skype.Detection',
+        'TDCCtl.TDCCtl',
+        'WMPlayer.OCX', // Windows media player
+        'rmocx.RealPlayer G2 Control',
+        'rmocx.RealPlayer G2 Control.1'
+      ]
         // starting to detect plugins in IE
-        result = this.map(names, function (name) {
-          try {
+      result = map(names, function (name) {
+        try {
             // eslint-disable-next-line no-new
-            new window.ActiveXObject(name)
-            return name
-          } catch (e) {
-            return null
-          }
-        })
-      }
-      if (navigator.plugins) {
-        result = result.concat(this.getRegularPlugins())
-      }
-      return result
-    },
-    pluginsShouldBeSorted: function () {
-      var should = false
-      for (var i = 0, l = this.options.sortPluginsFor.length; i < l; i++) {
-        var re = this.options.sortPluginsFor[i]
-        if (navigator.userAgent.match(re)) {
-          should = true
-          break
+          new window.ActiveXObject(name)
+          return name
+        } catch (e) {
+          return ERROR
         }
+      })
+    } else {
+      result.push(NOT_AVAILABLE)
+    }
+    if (navigator.plugins) {
+      result = result.concat(getRegularPlugins(options))
+    }
+    return result
+  }
+  var pluginsShouldBeSorted = function (options) {
+    var should = false
+    for (var i = 0, l = options.sortPluginsFor.length; i < l; i++) {
+      var re = options.sortPluginsFor[i]
+      if (navigator.userAgent.match(re)) {
+        should = true
+        break
       }
-      return should
-    },
-    touchSupportKey: function (keys) {
-      if (!this.options.excludeTouchSupport) {
-        keys.addPreprocessedComponent({key: 'touch_support', value: this.getTouchSupport()})
-      }
-      return keys
-    },
-    hardwareConcurrencyKey: function (keys) {
-      if (!this.options.excludeHardwareConcurrency) {
-        keys.addPreprocessedComponent({key: 'hardware_concurrency', value: this.getHardwareConcurrency()})
-      }
-      return keys
-    },
-    hasSessionStorage: function () {
-      try {
-        return !!window.sessionStorage
-      } catch (e) {
-        return true // SecurityError when referencing it means it exists
-      }
-    },
-    // https://bugzilla.mozilla.org/show_bug.cgi?id=781447
-    hasLocalStorage: function () {
-      try {
-        return !!window.localStorage
-      } catch (e) {
-        return true // SecurityError when referencing it means it exists
-      }
-    },
-    hasIndexedDB: function () {
-      try {
-        return !!window.indexedDB
-      } catch (e) {
-        return true // SecurityError when referencing it me
```

Comment:
This needs to capitalize components_key if we want to be a bit more backwards compatible.
Not sure why github shows that line; should be line 1343